client-sdk/go/modules/contracts/oas20/oas20.go

```diff
@@ -93,3 +130,157 @@ type Response struct {
 type BalanceResponse struct {
 	Balance quantity.Quantity `json:"balance"`
 }
+
+// EventDecoder is the OAS20 contract events decoder.
+type EventDecoder struct {
+	InstanceID contracts.InstanceID
+}
+
+// DecodeEvent implements client.EventDecoder.
+func (ed *EventDecoder) DecodeEvent(event *types.Event) (client.DecodedEvent, error) {
+	if !strings.HasPrefix(event.Module, contracts.ModuleName+".") {
```

Comment:
Could also take a code ID and only decode events emitted by that code?

Reply:
Yeah i think that makes sense. In the worst case, the user might need to do one more query to get the code ID in case they only have the instance ID.